File: Frontend/src/types/ticket.ts
export interface Ticket {
  id: string;
  title: string;
  description: string;
  status: 'Open' | 'In Progress' | 'Resolved' | 'Closed';
  customerEmail: string;
  resolution?: string;
  createdAt: string;
  updatedAt?: string;
  summary?: string;
}

export interface CreateTicketInput {
  title: string;
  description: string;
  customerEmail: string;
}

export interface UpdateTicketInput {
  status?: string;
  resolution?: string;
}

File: Backend/Entities/Ticket.cs
namespace TicketSystem.Entities;

public class Ticket
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Status { get; set; } = "Open";
    public string CustomerEmail { get; set; } = string.Empty;
    public string? Resolution { get; set; }
    public string? Summary { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
